cmake/ConfigureCompiler.cmake

```
@@ -7,6 +7,7 @@ set(WITH_UNDODB OFF CACHE BOOL "Use rr or undodb")
 set(FDB_RELEASE OFF CACHE BOOL "This is a building of a final release")
 
 add_compile_options(-DCMAKE_BUILD)
+add_compile_options(-DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_SYSTEM_NO_DEPRECATED)
```

This is not how this should be added. I realize that most of the definitions are added that way (and I realize that it was me who made this mistake in the first place), but cmake provides a `add_compile_definitions` function for this purpose. The problem with this is, that it is not portable.

```suggestion
add_compile_definitions(BOOST_ERROR_CODE_HEADER_ONLY BOOST_SYSTEM_NO_DEPRECATED)
```
We probably should change all other occurrences of this. Furthermore I realized that `add_definitions` was deprecated.

---

Thanks.  I know next to nothing about cmake so I was just following the existing style.

---

yaeh - this was definitely my mistake not yours. I will make a PR to change the other occurrences of this later